slapo/pipeline.py

```diff
@@ -425,6 +426,10 @@ def build_pipeline_model(sch, target, **kwargs):
             )
         )
 
+    # Transfer hooks from the original module to the corresponding pipeline stage.
+    transfer_hooks(res_partition[0], partitioned_mod, ["fwd_pre", "bwd_post"])
```

Comment:
shouldn't we transfer all the hooks? why just specify "fwd_pre", "bwd_post" for res_partition[0]? and what bwd_post is need to be transferred?

Reply:
Note that the original hook is registered to the top module that includes pipeline modules. Now we get rid of the top module and only take its children to be pipeline stages. Thus, the fwd_pre and bwd_post hooks should be transferred to the first submodule; while fwd_post should be transferred to the last submodule.